rollup.config.js

```diff
@@ -108,18 +108,22 @@ function createCommonJSConfig(input, output, options) {
 }
 
 function createUMDConfig(input, output, env) {
-  const c = output.split('/').pop()
+  const c = output.replace(/^dist\/umd\//, '').split(/\W/)
+  const name = c.reduce((acc, itm, idx) => {
```

Comment:
I think you can extract the function inside `reduce`,
so that it's a named function. It can be extracted to a `const` inside this scope.

Something like:
```js
const toCamelCase = (acc, item, i) => { /* ... */ }
const name = c.reduce(toCamelCase, 'zustand')
```

Reply:
I don't think it helps much, but how would you name it? `toCamelCase` doesn't sound correct for `(acc, item, i) => ...`.

Reply:
I don't know the intent of the function, but from what I've read below in the implementation, it makes this: `['happy', 'hippo']` to this: `'zustandHappyHippo'`, right?

```js
['happy', 'hippo'].reduce((acc, itm, idx) => {
  if (idx === 2 - 1 && itm === 'index') {
    return acc
  }
  return acc + `${itm.slice(0, 1).toUpperCase()}${itm.slice(1)}`
}, 'zustand')

// => 'zustandHappyHippo'
```

That's why I named it `toCamelCase`. You can name it `toCamelCaseWithPrefix` as well.

Reply:
I know what you mean, but it doesn't sound like a reducer function. Maybe, `toCamelCaseReducer`.

Reply:
I'll be a bit philosophical now, so excuse me for that :P 
For me, `toCamelCaseReducer` would make sense if such function
was exported as a public interface.
Here, in the local block scope, `list.reduce(toCamelCaseWithPrefix, 'zustand')`
reads like a sentence, that's why I chose it.

With that said, `toCamelCaseReducer` is a better name than no name at all,
so you can go with it :)

Reply:
To continue a bit philosophical discussion, what I don't prefer is
```js
const toCamelCase = (acc, itm, idx) => ...
```
which doesn't make sense about what are acc, itm, and even idx. I'd expect
```js
const toCameCase = (str, maybeSomethingRelatedWithString) => ...
```

Reply:
I see your point. I'm used to TDD-style design, so if I have to,
I will trade the readability of the usage, even if the implementation
or declaration is less readable. In this context, my idea is more
readable _for me and my habits, in this context alone_,
but I perfectly understand where are you going from, so I think
your idea is as good, so go for it :)